### PR TITLE
Ryan Stemmle BE 2102 WEEK 3 IC

### PR DESCRIPTION
**Pull request by r-stemmle**
Ryan Stemmle BE 2102 WEEK 3 IC

### CODE REVIEW

File: lib/building.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @units.find_all do |unit|
+      unit.renter != nil
+    end
+  end
+
+  def renters
+    units.map do |unit|
+      unit.renter.name
+    end
+  end
```
Comment (mikedao) [inline lib/building.rb:47]: This does not work if there is a vacancy

File: lib/building.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    units_hash
+  end
+
+  def renter_with_highest_rent
+    highest = rented_units[-1].renter
+    rented_units.sort_by do |unit|
+      unit.monthly_rent
+    end
+    highest = rented_units[-1].renter
+  end
```
Comment (mikedao) [inline lib/building.rb:29]: max_by would have been a better choice here.

File: test/apartment_test.rb [hunk 3]
```diff
@@ -4,4 +4,48 @@
...
+
+  def test_it_exists
+    @unit1
+
+    assert_instance_of Apartment, @unit1
+  end
+
+  def test_it_has_apt_number
+    @renter1
+    @unit1
```
Comment (mikedao) [inline test/apartment_test.rb:20]: You should not have to call instance variuables in your test in this fashion

**Review by mikedao — COMMENTED** [review]
Good work overall, watch your edge cases.

Score: 3
